libcodechecker/server/client_db_access_handler.py

```diff
@@ -344,64 +335,25 @@ def getRunResultCount(self, run_id, report_filters):
                                               msg)
 
     @timeit
-    def __construct_bug_event_list(self, session, start_bug_event):
+    def __construct_bug_item_list(self, session, report_id, item_type):
 
-        file_path_cache = {}
-        bug_events = []
-        event = session.query(BugPathEvent).get(start_bug_event)
-
-        file_path = file_path_cache.get(event.file_id)
-        if not file_path:
-            f = session.query(File).get(event.file_id)
-            file_path = f.filepath
-            file_path_cache[event.file_id] = file_path
-
-        bug_events.append((event, file_path))
+        q = session.query(item_type) \
+            .filter(item_type.report_id == report_id) \
+            .order_by(item_type.order)
 
-        while event.next is not None:
-
-            event = session.query(BugPathEvent).get(event.next)
+        file_path_cache = {}
+        bug_items = []
 
+        for event in q:
             file_path = file_path_cache.get(event.file_id)
             if not file_path:
                 f = session.query(File).get(event.file_id)
```

Comment:
I wonder, do we need a cache here? Wouldn't SQLAlchemy's ORM already do caching?
See http://docs.sqlalchemy.org/en/latest/orm/session_basics.html#is-the-session-a-cache